src/main/java/seedu/address/logic/commands/ListAppointmentsCommand.java

```diff
@@ -58,12 +60,36 @@ public CommandResult execute(Model model) {
         LocalDateTime now = LocalDateTime.now();
         Optional<LocalTime> effectiveTimeFilter = dateFilter.isPresent() ? timeFilter : Optional.empty();
 
-        List<Person> filteredAppointments = personList.stream()
-                .filter(person -> person.hasAppointment(now, dateFilter, effectiveTimeFilter))
-                .collect(Collectors.toList());
+        List<AppointmentInfo> allAppointments = new ArrayList<>();
 
-        return new CommandResult(String.format(MESSAGE_SUCCESS, filteredAppointments.size())
-                + "\n" + formatAppointments(filteredAppointments));
+        for (Person person : personList) {
+            for (Schedule schedule : person.getSchedules()) {
+                LocalDateTime appointmentDateTime = LocalDateTime.parse(schedule.getDateTime(),
+                        DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
+
+                if (appointmentDateTime.isBefore(now)) {
```

Comment:
Could the inner loop logic maybe be extracted into a separate method to reduce the nesting level?